# Calculate format=diff merge(media-gfx/darktable)!=2.6.0&&os_install_arch_machine==i686
diff --git a/src/iop/filmic.c b/src/iop/filmic.c
index fd6a176..99096ca 100644
--- a/src/iop/filmic.c
+++ b/src/iop/filmic.c
@@ -584,7 +584,7 @@ void process_sse2(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, c
 
       }
 
-      rgb = luma + _mm_set1_ps(concavity) * (rgb - luma);
+      rgb = _mm_set1_ps(luma) + _mm_set1_ps(concavity) * (rgb - _mm_set1_ps(luma));
       rgb = _mm_max_ps(rgb, zero);
       rgb = _mm_min_ps(rgb, one);
 
